ui/src/app/pages/homepage/registration-dashboard.tsx

```diff
@@ -91,12 +91,21 @@ function redirectToTwoFactorSetup(): void {
 
 function redirectToNiH(): void {
   AnalyticsTracker.Registration.ERACommons();
-  const url = serverConfigStore.getValue().shibbolethUiBaseUrl + '/login?return-url=' +
```

Comment:
Looks like accidental RAS prototyping changes in the commit - please revert

Reply:
wow. thanks for letting me know. RAS was in my another application window. I didn't realize UI and API is in the same github branch. 
I almost put client id/secret there. lucky. 

Side note, why we want to have all code in the same repo?

Reply:
Short version: single repo is simpler to manage, e.g. just for git flow purposes and for managing/testing UI / API changes across both. For this project, the API server is really a "frontend" server (a wrapper around Terra with some database storage) that exists for the purposes of supporting the UI, so they are tightly linked.

Managing multiple synchronized repo's tends to be a pain in git, and lacks basic tooling to do this well in my experience. That said, I'm not entirely opposed to splitting this out if there's a proposal for doing this well and the benefits can be shown.

Reply:
The UI and API are separate servers, in different binaries. The API is not a real 'infra' API server, but that is still another standalone server that is different from the UI server. 

- As team grows, and both repo grows, I would image it is harder for 50 people working on the same PR, and 300+ Circle job running in parallel 
- We may want those two servers managed/versioned separately?  We don't want a PR that breaks API always triggered random UI rollback together.  
- I am not familiar with how that makes git flows easier, is that because we can always git checkout the entire codebase and run PR test against it? I buy that, that is slightly better than post-submit integration test? Actually I thought the e2e test for us is just puppetest, and that one we uses aou-test, not local server....
- No strong preference, 10-20 people is still a reasonable size to write code in the same repo daily...

Reply:
> As team grows, and both repo grows, I would image it is harder for 50 people working on the same PR, and 300+ Circle job running in parallel

Agreed, if we wanted to scale to that size it would probably make sense to change things operationally. As of today, we have several small teams working vertically on the stack, so I don't see an argument for this increasing efficiency. If we did instead split the team ownership by UI vs API, we would definitely want to more seriously consider this option. 

> We may want those two servers managed/versioned separately? We don't want a PR that breaks API always triggered random UI rollback together.

They can already be deployed or rolled back at different versions. I'm not aware of any issues we have in our current deployment infrastructure that would be resolved by splitting into two repos.

> I am not familiar with how that makes git flows easier, is that because we can always git checkout the entire codebase and run PR test against it? I buy that, that is slightly better than post-submit integration test? Actually I thought the e2e test for us is just puppetest, and that one we uses aou-test, not local server....

If you're developing the API, you likely need to run the UI server. Now you to have two branches which are just manually synchronized. If someone merges changes, you'll need to think about which repo you need to rebase or not. Any automated mechanism for expressing this dependency is usually pretty bad in git, e.g. submodules. I've seen git subtree as well which is suppsedly an improvement, but I haven't tried it and I think it's still more complicated than the mono repo approach we have now.

Puppeteer test would ideally start a local hermetic API server. I hope we can do this someday which should further decrease flakiness of our integration tests.